src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -1443,31 +1441,12 @@ void MacroAssembler::check_klass_subtype_fast_path(Register sub_klass,
   // So if it was a primary super, we can just fail immediately.
   // Otherwise, it's the slow path for us (no success at this point).
 
-  if (super_check_offset.is_register()) {
-    br(Assembler::EQ, *L_success);
-    subs(zr, super_check_offset.as_register(), sc_offset);
-    if (L_failure == &L_fallthrough) {
-      br(Assembler::EQ, *L_slow_path);
-    } else {
-      br(Assembler::NE, *L_failure);
-      final_jmp(*L_slow_path);
-    }
-  } else if (super_check_offset.as_constant() == sc_offset) {
-    // Need a slow path; fast failure is impossible.
-    if (L_slow_path == &L_fallthrough) {
-      br(Assembler::EQ, *L_success);
-    } else {
-      br(Assembler::NE, *L_slow_path);
-      final_jmp(*L_success);
-    }
+  sub(rscratch1, super_check_offset, in_bytes(Klass::secondary_super_cache_offset()));
+  if (L_failure == &L_fallthrough) {
+    cbz(rscratch1, *L_slow_path);
```

Comment:
What's the reason for changing to cbz/cbnz from subs and branching on condition codes?

Reply:
When I was porting this code to AArch64, I mostly used x86_64 as my reference. In order to minimize the number of errors, I avoided AArch64 instructions such as `cbz`. That's no longer an issue, so we can use AArch64 idioms like this, which avoid the x86 problem of accidentally clobbering flags by not using flags except where necessary. I don't think there's a performance difference.